tests/old_suite/compat/test_multipackage.py

```diff
@@ -0,0 +1,34 @@
+import os
+import sys
+
+
+class Struct(object):
+    def __init__(self, entries):
+        self.__dict__.update(entries)
+
+opts = {
+    'all_with_crypto': False,
+    'clean': False,
+    'interactive_tests': False,
+    'verbose': True,
+    'run_known_fails': True,
+}
+
+args = []
+for file in os.listdir(os.path.join(os.path.dirname(os.path.dirname(__file__)), 'multipackage')):
+    if file.endswith('.py'):
+        args.append(os.path.join('multipackage', file))
+
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+
+import runtests
+
+for test in sorted(runtests.suite(Struct(opts), args)._tests, key=lambda test: test._testMethodName):
+    test_name = str(test).split(' ')[0].title()
+    TestClass = type(test_name, (type(test),), test.__dict__)
+    TestClass._testcase = None
+    exec('TestClass.{} = test._generic_test_function'.format(test._testMethodName))
+    exec('{} = TestClass'.format(test_name))
+    del TestClass
+
+del test
```

Comment:
This looks like it could be done easier using py.test mechanisms. They are a bit complicated to learn, but worth it :-)